src/ompl/geometric/planners/experience/src/ThunderRetrieveRepair.cpp

```diff
@@ -0,0 +1,456 @@
+        //repair_planner->setGoalBias(0.2);
+
+        OMPL_DEBUG("No repairing planner specified. Using default: %s", repair_planner->getName().c_str() );
+        repairPlanner_ = repair_planner; //ompl::geometric::Planner( repair_planer );
+    }
+    // Setup the problem definition for the repair planner
+    repairProblemDef_->setOptimizationObjective(pdef_->getOptimizationObjective()); // copy primary problem def
+
+    // Setup repair planner
+    repairPlanner_->setProblemDefinition(repairProblemDef_);
+    if (!repairPlanner_->isSetup())
+        repairPlanner_->setup();
+
+
+}
+
+void ompl::geometric::ThunderRetrieveRepair::freeMemory(void)
+{
+}
+
+ompl::base::PlannerStatus ompl::geometric::ThunderRetrieveRepair::solve(const base::PlannerTerminationCondition &ptc)
+{
+    bool solved = false;
+    double approxdif = std::numeric_limits<double>::infinity();
+    nearestPaths_.clear();
+
+    // Check if the database is empty
+    if (experienceDB_->isEmpty())
+    {
+        std::cout << ANSI_COLOR_GREEN;
+        OMPL_INFORM("Experience database is empty so unable to run ThunderRetrieveRepair algorithm.");      
+        std::cout << ANSI_COLOR_RESET;
+
+        // TODO: it seems TIMEOUT causes the wrong behavior in parallel plan
+        return base::PlannerStatus::CRASH;
+    }
+
+    // Get a single start and goal state TODO: more than one
+    const base::State *startState = pis_.nextStart();
+    const base::State *goalState = pis_.nextGoal(ptc);
+
+    // Create solution path struct
+    ompl::geometric::SPARSdb::CandidateSolution candidateSolution;
+
+    // Search for previous solution in database
+    // TODO make this more than 1 path
+    if (!experienceDB_->findNearestStartGoal(nearestK_, startState, goalState, candidateSolution, ptc))
+    {
+        std::cout << ANSI_COLOR_RED;
+        OMPL_INFORM("RetrieveRepair::solve() No nearest start or goal found");        
+        std::cout << ANSI_COLOR_RESET;
+        return base::PlannerStatus::TIMEOUT; // The planner failed to find a solution
+    }
+
+    
+    // Save this for future debugging
+    nearestPaths_.push_back(candidateSolution.getGeometricPath());
+    nearestPathsChosenID_ = 0; // TODO not hardcode
+
+    // All save trajectories should be at least 2 states long, then we append the start and goal states, for min of 4
+    assert(candidateSolution.getStateCount() >= 4);
+
+    // Smooth the result
+    /* TODO re-enable
+    OMPL_INFORM("ThunderRetrieveRepair solve: Simplifying solution (smoothing)...");
+    time::point simplifyStart = time::now();
+    std::size_t numStates = primaryPath.getStateCount();
+    psk_->simplify(*primaryPath, ptc);
+    double simplifyTime = time::seconds(time::now() - simplifyStart);
+    OMPL_INFORM("ThunderRetrieveRepair: Path simplification took %f seconds and removed %d states", simplifyTime, numStates - primaryPath.getStateCount());
+    */
```

Comment:
Does it make sense to do this? Make this an option of the algorithm, rather than have it as commented code?